src/idl_gen_general.cpp

```diff
@@ -203,8 +203,13 @@ class GeneralGenerator : public BaseGenerator {
       const std::string &classcode, bool needs_includes) {
     if (!classcode.length()) return true;
 
-    std::string code;
-    code = code + "// " + FlatBuffersGeneratedWarning();
+    std::string code("//------------------------------------------------------------------------------\n"
```

Comment:
No need for the extra `-----------------`, just add the `// <auto-generated>\n`.
You can still call `FlatBuffersGeneratedWarning();` on the next line.
Also, I believe https://github.com/google/flatbuffers/issues/4287 showed that there should be a `</auto-generated>` at the end of the file?